### PR TITLE
Hub reporting drop satellite info

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

Prepare for registering a Server in the Hub
- drop rhnSatelliteInfo
- introduce suseMgrServerInfo
- provide a grains module to detect a MgrServer and provide reportdb settings.
- provide state module to create db user
- store the mgrServer settings in the new database table
- create the DB user on bootstrapping

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- Unit tests were added

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/16566

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/suse/manager/reactor/messaging/ApplyStatesEventMessage.java [hunk 1]
```diff
@@ -123,6 +155,15 @@ public Long getServerId() {
         return stateNames;
     }
 
+    /**
+     * Return the state specific pillar data
+     *
+     * @return the pillar data
+     */
+    public Map<String, Object> getPillar() {
```
Comment (lucidd) [inline java/code/src/com/suse/manager/reactor/messaging/ApplyStatesEventMessage.java:164]: Can we directly return Optional here or would returning an empty map also make sense?

File: java/code/src/com/suse/manager/reactor/messaging/RegistrationUtils.java [hunk 2]
```diff
@@ -148,6 +153,27 @@ public static void finishRegistration(MinionServer minion, Optional<ActivationKe
         if (applyHighstate) {
             MessageQueue.publish(new ApplyStatesEventMessage(minion.getId(), true, emptyList()));
         }
+        // Create a report db user when system is a mgr server
+        if (minion.isMgrServer()) {
+            // create default user with random password
+            MgrServerInfo mgrServerInfo = minion.getMgrServerInfo();
+            Credentials credentials = CredentialsFactory.createCredentials(
+                    "hermes_" + RandomStringUtils.random(8, "abcdefghijklmnopqrstuvwxyz"),
+                    RandomStringUtils.randomAlphanumeric(20),
```
Comment (lucidd) [inline java/code/src/com/suse/manager/reactor/messaging/RegistrationUtils.java:162]: if i see this correctly RandomStringUtils is using Random internationally which is not supposed to be used for any security related things like passwords.

File: java/code/src/com/suse/manager/reactor/messaging/ApplyStatesEventMessage.java (hunk 1 above)
Comment (mcalmer) [reply java/code/src/com/suse/manager/reactor/messaging/ApplyStatesEventMessage.java]: done

File: java/code/src/com/suse/manager/reactor/messaging/RegistrationUtils.java (hunk 2 above)
Comment (mcalmer) [reply java/code/src/com/suse/manager/reactor/messaging/RegistrationUtils.java]: made a bit different implementation and use SecureRandom()
I think this should be sufficient.

File: spacewalk/certs-tools/spacewalk-certs-tools.changes [hunk 3]
```diff
@@ -1,3 +1,6 @@
+- delopy local CA under different name in the truststore to avoid
```
Comment (mackdk) [inline spacewalk/certs-tools/spacewalk-certs-tools.changes:1]: Typo "delopy" here and in the other changelog

**Review by mackdk — COMMENTED** [review]
Looks good to me, just a couple of minor comments.

**Review by mbussolotto — APPROVED** [review]
Besides "delopy"  typo, seems fine

File: java/Makefile.docker [hunk 4]
```diff
@@ -7,6 +7,8 @@ DOCKER_REGISTRY       = registry.opensuse.org
 DOCKER_VOLUMES        = -v "$(CURDIR)/../:/manager"
 DOCKER_PG_PORTS       = -p 5432:5432
 DOCKER_COMMON_OPTS    = --interactive --tty --rm
+#DOCKER_ENV            = -e NEXTVERSION='4.3.10'
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply java/Makefile.docker:10]: This is only needed when you want to migrate to a specific schema version. I needed that as the current was still 4.3.8 and next was calculated to 4.3.9. But we had our migrations already in 9-to10.

File: schema/spacewalk/postgres/packages/rhn_channel.pkb [hunk 5]
```diff
@@ -303,12 +303,12 @@ update pg_settings set setting = 'rhn_channel,' || setting where name = 'search_
                 select  1
                 from    rhnChannelFamily
                 where   id = channel_family_id_in
-                    and label = 'rhn-proxy';
+                    and label = 'SMP';
```
[the remark this answers is not in the document]
Comment (mcalmer) [reply schema/spacewalk/postgres/packages/rhn_channel.pkb:306]: No. rhn-proxy and rhn-satellite were only used for original satellite. I just "fixed" something which is broken for years.

File: spacewalk/certs-tools/spacewalk-certs-tools.changes (hunk 3 above)
Comment (mcalmer) [reply spacewalk/certs-tools/spacewalk-certs-tools.changes]: fixed

File: java/Makefile.docker (hunk 4 above)
[the remark this answers is not in the document]
Comment (mcalmer) [reply java/Makefile.docker:11]: added a comment

**Review by mackdk — APPROVED** [review]

**Pull request merged**